tutorials/video_classification_example/data.py

```diff
@@ -0,0 +1,286 @@
+                    Normalize((args.audio_logmel_mean,), (args.audio_logmel_std,)),
+                ]
+            ),
+        )
+
+    def _make_ds_and_loader(self, mode: str):
+        """Creates both the dataset and dataloader for a given dataset split 'mode'. This returns
+        both the dataset and the dataloader specified, and should be called from self.{train|val|test}_dataloader().
+
+        Args:
+            mode (str): The dataset split to create. Should be 'train' or 'val'.
+        """
+        ds = LimitDataset(
+            labeled_video_dataset(
+                data_path=str(
+                    Path(self.root)
+                    / (self.TRAIN_PATH if mode == "train" else self.VAL_PATH)
+                ),
+                clip_sampler=make_clip_sampler(
+                    "random" if mode == "train" else "uniform", self.args.clip_duration
+                ),
+                video_path_prefix=self.args.video_path_prefix,
+                transform=self._make_transforms(mode=mode),
+                video_sampler=DistributedSampler
+                if (self.trainer is not None and self.trainer.use_ddp)
+                else RandomSampler,
+            )
+        )
+        return ds, DataLoader(
+            ds, batch_size=self.args.batch_size, num_workers=self.args.workers
+        )
+
+    def train_dataloader(self):
+        self.train_dataset, loader = self._make_ds_and_loader("train")
+        return loader
+
+    def val_dataloader(self):
+        self.val_dataset, loader = self._make_ds_and_loader("val")
+        return loader
+
+
+class LimitDataset(torch.utils.data.Dataset):
+    """
+    To ensure a constant number of samples are retrieved from the dataset we use this
+    LimitDataset wrapper. This is necessary because several of the underlying videos
+    may be corrupted while fetching or decoding, however, we always want the same
+    number of steps per epoch.
+    """
+
+    def __init__(self, dataset):
+        super().__init__()
+        self.dataset = dataset
+        self.dataset_iter = itertools.chain.from_iterable(
+            itertools.repeat(iter(dataset), 2)
+        )
+
+    def __getitem__(self, index):
+        return next(self.dataset_iter)
+
+    def __len__(self):
+        return self.dataset.num_videos
+
+
+class KineticsDataModule(LabeledVideoDataModule):
+    TRAIN_PATH = "train.csv"
+    VAL_PATH = "val.csv"
+    NUM_CLASSES = 700
+
+
+class MiniKineticsDataModule(LabeledVideoDataModule):
+
+    TRAIN_PATH = "train"
+    VAL_PATH = "val"
+    SOURCE_URL = "https://pl-flash-data.s3.amazonaws.com/kinetics.zip"
+    SOURCE_DIR_NAME = "kinetics"
+    NUM_CLASSES = 6
+
+
+class UCF11DataModule(LabeledVideoDataModule):
+    TRAIN_PATH = None
+    VAL_PATH = None
+    SOURCE_URL = "https://www.crcv.ucf.edu/data/YouTube_DataSet_Annotated.zip"
+    SOURCE_DIR_NAME = "action_youtube_naudio"
+    NUM_CLASSES = 11
+
+    def __init__(self, args):
+        args.verify = False
```

Comment:
It's a bit hacky setting the args parameter throughout your program, imo it should be a immutable constant that aligns with the args that were sent into the program. Can you add verify as another parameter in LabeledVideoDataModule? Then just pass it in? Also would be fine with it hardcoded to false for all datasets, not too concerned about the ssl certificates tbh.

Reply:
Got rid of args